### DISCUSSION TITLE
Combined link-generator + parser

### DISCUSSION THREAD

**Discussion by linas** [opencog/link-grammar#1523] [Q&A]
@ampli what are the technical issues that would block a combined link-generator+link-parser? So that for example) if `link-parser` is started with a `--generate` flag, then it would parse the way it does today, except when it sees `\*` in the input stream; then it would treat this as a generator wildcard, but otherwise continue as normal?

This would vaguely resemble what the spell-guesser does. ...

**Comment by ampli** [comment]
There is no need for a `--generate` flag, as the library can set the generation mode for the current sentence if any generation syntax is detected. (Generation syntax may include `\*` and any additional syntax we may add, e.g., `\[word1, word2]` for a list of words, or `\{}` over optional words.) The generation backend (we still need it in order to stuff words instead of categories) would be part of `link-parser` and be invoked at the final stage if the sentence got parsed in generation mode (we may have a read-only parse option to indicate that). We would have to add some of the `link-generator` flags as `link-parser` variables and merge its manual with `link-parser`.

**Reply by linas** [reply]
OK, cool. I plan to mess with generation in the upcoming weeks/months. Mostly at the API level; having the command-line tool helps as a sanity-check.
